### PR TITLE
Golang Getting started - based on roll dice service and Automatic Ins…

### PR DESCRIPTION
**Pull request by pegasas**
try to add an roll dice example with golang, solving https://github.com/open-telemetry/opentelemetry.io/issues/2623

### CODE REVIEW

**Review by MrAlias — CHANGES_REQUESTED** [review]
This changes the substance of this documentation from manual instrumentation to automatic instrumentation. Automatic instrumentation deserves to live besides this not in-place of it.

**Comment by svrnm** [comment]
thanks for raising this @pegasas. This still looks work-in-progress for me, so please put this into draft mode for now.

> This changes the substance of this documentation from manual instrumentation to automatic instrumentation. Automatic instrumentation deserves to live besides this not in-place of it.

Right now I don't see how the documentation has been changed from manual to automatic instrumentation, as it seems incomplete. To comment on that nevertheless, as we have a discussion on the same ongoing [over here](https://github.com/open-telemetry/opentelemetry.io/issues/3228): _ the assumption is that a Getting Started should give you something tangible as quick as possible in your preferred language_, so if the Golang SIG at some point comes to the conclustion that automatic instrumentation might be a better route to that, we can adopt it in the Getting Started, and have a similar structure as we have it for Java & JavaScript these days (Getting Started using mostly automation, have a "manual" and "automatic" documentation).

File: content/en/docs/instrumentation/go/getting-started.md [hunk 1]
```diff
@@ -4,160 +4,79 @@ weight: 10
...
-
-To start building the application, make a new directory named `fib` to house our
-Fibonacci project. Next, add the following to a new file named `fib.go` in that
-directory[^1].
-
-[^1]:
-    The `Fibonacci()` function intentionally produces invalid results for
-    sufficiently large values of `n`. This is addressed in
-    [Error handling](#error-handling).
+This page will show you how to get started with OpenTelemetry in Golang.
```
Comment (pellared) [inline content/en/docs/instrumentation/go/getting-started.md:7]: The language is called "Go". Not "Golang" 😉

File: content/en/docs/instrumentation/go/getting-started.md [hunk 2]
```diff
@@ -4,160 +4,79 @@ weight: 10
...
-import (
-	"context"
-	"fmt"
-	"io"
-	"log"
-)
+The following example uses a basic
+[Gin Web Framework](https://gin-gonic.com/docs/introduction/). If you are
+not using Gin, that's OK — you can use OpenTelemetry Golang with other web
+frameworks as well, such as Goframe and Beego.
```
Comment (pellared) [inline content/en/docs/instrumentation/go/getting-started.md:24]: Can we simply use `net/http`? You can mention that it works for Gin, Gorilla/Mux, Echo etc.

File: content/en/docs/instrumentation/go/getting-started.md [hunk 3]
```diff
@@ -4,160 +4,79 @@ weight: 10
...
+that directory, create a file called `build.gradle.kts` with the following
+content:
 
-// Poll asks a user for input and returns the request.
-func (a *App) Poll(ctx context.Context) (uint, error) {
-	a.l.Print("What Fibonacci number would you like to know: ")
+```go
+go mod init
+go get github.com/gin-gonic/gin
+go mod vendor
```
Comment (pellared) [inline content/en/docs/instrumentation/go/getting-started.md:42]: Vendoring is not needed.

**Comment by pellared** [comment]
> I will push out again when I resolve automatic instrumentation.

What do you mean? Right now, getting started should work with manual instrumentation.

**Comment by cartermp** [comment]
So I think we should change this eventually to use autoinstrumentation first, but the main issue is that the Go autoinstrumentation agent is still quite early. I don't know if we should endorse it as the right way to get started today.

**Comment by pegasas** [comment]
> So I think we should change this eventually to use autoinstrumentation first, but the main issue is that the Go autoinstrumentation agent is still quite early. I don't know if we should endorse it as the right way to get started today.

https://github.com/open-telemetry/opentelemetry-go-instrumentation
Indeed, 
It seems this project is progressing.
Is this the only project which adapted eBPF?

**Comment by cartermp** [comment]
Yes, for now it is. There may be more in the future though, since the technique could also apply for Rust and C++. But I don't know if there's any motivation on those SIGs to start an autoinstrumentation project.

**Comment by pellared** [comment]
> Is this the only project which adapted eBPF?

There is also https://github.com/open-telemetry/opentelemetry-ebpf

**Comment by svrnm** [comment]
@pegasas there is now an updated getting started for go using _manual_ instrumentation. Do you still have interest in writing something about automatic instrumentation or can we close this PR?

**Comment by pegasas** [comment]
sure. @svrnm .

I close this PR.
I am interested in other items in automatic instrumentation.
If there is any. feel free to assign to me.

**Comment by svrnm** [comment]
@pegasas my question was around having some section added to the go lang documentation around automatic instrumentation. #3345 is not about Go and not about Automatic instrumentation, can you provide more details?

**Pull request closed without merging**